### DISCUSSION TITLE
PowerShell ProgressBar -- Part 4 – Tiberriver256

### DISCUSSION THREAD

**Discussion by giscus** [Tiberriver256/Tiberriver256.GitHub.io#13] [General]
# PowerShell ProgressBar -- Part 4 – Tiberriver256

XAML Templates -- Material Design.

https://tiberriver256.github.io/powershell/PowerShellProgress-Pt4/

**Comment by afisher-cont** [comment]
This was VERY helpful, one thing while building my own bar, I would love to be able to change the actual Background color of the window. I see there is a "Dark Theme" that does this, but is it possible to change it to an arbitrary color?

**Reply by Tiberriver256** [reply]
Hello 👋,

Glad you enjoyed the blog series. If you're working on building your own custom XAML progress bar the attribute you're looking for is `Background` on the `Window` element. I have mine set in this post to `{DynamicResource MaterialDesignPaper}` which means it gets the value from MaterialDesignPaper but you could change it to something like `Blue` if you wanted.

The valid plain English colors you can use are here:
https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.brushes?view=windowsdesktop-7.0

You can also do your own color variants using various methods laid out here:
 https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.solidcolorbrush?view=windowsdesktop-7.0#xaml-attribute-usage

If you're looking at doing it with my module, I don't currently have that functionality built into it but would definitely accept a PR 😀
